### ISSUE TITLE
Direct sort expressions do not work

### ISSUE THREAD

**Issue by mikehaertl** [yiisoft/yii2#16314]
### What steps will reproduce the problem?

Create e.g. an ActiveDataProvider and use a custom sort expression as documented in `Sort::$attributes`:

> Since 2.0.12 particular sort direction can be also specified as direct sort expression, like following:                
>  ```php                                                                                                                 
> 'name' => [
>     'asc' => '[[last_name]] ASC NULLS FIRST', // PostgreSQL specific feature
>     'desc' => '[[last_name]] DESC NULLS LAST',
> ]
> ```

#### Example

```
$dataProvider = new \yii\data\ActiveDataProvider([
    'query' => \app\models\Job::find(),
    'sort' => [
        'attributes' => [
            'started_at' => [
                'asc' => '[[started_at]] IS NULL ASC, [[started_at]] ASC',
                'desc' => '[[started_at]] IS NULL DESC, [[started_at]] DESC',
                'default' => SORT_DESC,
                'name' => 'started',
            ],
            'created_at',
        ],
        'defaultOrder' => [
            'started_at' => SORT_DESC,
            'created_at' => SORT_DESC,
        ],
    ],
]);
```

### What is the expected result?

It should sort by NULL and then ascending.

### What do you get instead?

```
Database Exception – yii\db\Exception
SQLSTATE[42S22]: Column not found: 1054 Unknown column '0' in 'order clause'
The SQL being executed was: SELECT * FROM `job` ORDER BY `0`, `created_at` DESC LIMIT 20
Error Info: Array
(
    [0] => 42S22
    [1] => 1054
    [2] => Unknown column '0' in 'order clause'
)

Caused by: PDOException
SQLSTATE[42S22]: Column not found: 1054 Unknown column '0' in 'order clause'
in /var/www/vendor/yiisoft/yii2/db/Command.php at line 1258
```
#### Output of Sort::getOrders()

 ```
print_r($dataProvider->sort->getOrders());
// Array ( [0] => [[started_at]] IS NULL DESC, [[started_at]] DESC [created_at] => 3 )
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      |  7.1.17
| Operating system | Alpine Linux

**Comment by samdark** [comment]
Could be fixed in the next major version, I think. Do you want to do it and send us a pull request?

**Comment by mikehaertl** [comment]
Actually I already tried to find the problem but somehow got lost in the depth of the framework :). As far as I understand, `QueryTrait::orderBy()` does not accept raw sort strings. It makes me wonder how the above feature should work at all.

Is there a workaround? And can't this still be fixed in a 2.0.x bugfix release?

**Comment by rastikw** [comment]
@mikehaertl Use `\yii\db\Expression` (see [QueryTrait::addOrderBy](https://www.yiiframework.com/doc/api/2.0/yii-db-querytrait#addOrderBy()-detail)):

```
'started_at' => [
                'asc' => new \yii\db\Expression('[[started_at]] IS NULL ASC, [[started_at]] ASC'),
                'desc' => new \yii\db\Expression('[[started_at]] IS NULL DESC, [[started_at]] DESC'),
                'default' => SORT_DESC,
                'name' => 'started',
            ],
```
